go-get-version

```diff
@@ -1,8 +1,36 @@
 #!/bin/bash
 
-GO_VERSION="$(grep '^toolchain' $INPUT_MODFILE | sed 's/toolchain go*//')"
+GO_VERSION=""
+
+find_go_version() {
+    local modfile="$1"
+    local gv
+    gv="$(grep '^toolchain' $modfile | sed 's/toolchain go*//')"
+    if [ -z "$gv" ]; then
+        gv="$(grep '^go' $modfile | awk '{ print $2 }')"
+    fi
+    echo "$gv" 
+}
+
+# if INPUT_MODFILE is defined, attempt to read the go version from it
+if [ -n "$INPUT_MODFILE" -a -f "$INPUT_MODFILE" ]; then
+    GO_VERSION=$(find_go_version "${INPUT_MODFILE}")
+fi
+
+# if GO_VERSION is still not defined, attempt to find it in the default search paths
+if [ -z "$GO_VERSION" ]; then
+    DEFAULT_SEARCH_PATHS=". v1 v2 v3"
+
+    for p in $DEFAULT_SEARCH_PATHS; do
+        if [ -f "${p}/go.mod" ]; then
+            GO_VERSION=$(find_go_version "${p}/go.mod")
+            break;
+        fi
+    done
+fi
```

Comment:
I would change this so that default value for `INPUT_MODFILE` would be `./go.mod`. That way the default and custom case would be handled in first if case. That would also allow simplifying the loop to just `for i in {0..10}`.

Reply:
implemented in https://github.com/elisa-actions/setup-go-and-mage/pull/4/commits/e9a8c4f09757e7a0d9853cb9d350eeedf2622865